src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php

```diff
@@ -593,6 +595,87 @@ public function iEditItToHaveEmptyFixedDiscountInChannel(ChannelInterface $chann
         $this->formElement->specifyLastActionDiscountForChannel('', $channel);
     }
 
+    /**
+     * @When I filter by :channel channel
+     */
+    public function iFilterByChannel(ChannelInterface $channel): void
+    {
+        $this->filterElement->chooseChannel($channel);
+        $this->filterElement->filter();
+    }
+
+    /**
+     * @When I filter enabled catalog promotions
+     */
+    public function iFilterEnabledCatalogPromotions(): void
+    {
+        $this->filterElement->chooseEnabled();
+        $this->filterElement->filter();
+    }
+
+    /**
+     * @When /^I filter by (active|failed|inactive|processing) state$/
+     */
+    public function iFilterByState(string $state): void
+    {
+        $this->filterElement->chooseState(ucfirst($state));
+        $this->filterElement->filter();
+    }
+
+    /**
+     * @When I search by :phrase name
+     * @When I search by :phrase code
```

Comment:
I wonder isn't `I search by :phrase` enough 🤔

Reply:
No problem for UI, but in the case of API it'd necessitate choosing either or.

Reply:
👍